subprojects/core-api/src/main/java/org/gradle/api/configuration/BuildFeature.java

```diff
@@ -19,42 +19,59 @@
 import org.gradle.api.provider.Provider;
 
 /**
- * Status of a feature in a build that affects Gradle behavior,
- * and may impose additional requirements on plugins or build scripts.
+ * Status of a Gradle feature in the current invocation, such as Configuration Cache.
  * <p>
- * It is possible to check if the feature is {@link #getActive() active} in the current build.
- * The {@link #getRequested() requested} property shows whether the user opted in or opted out from the feature.
+ * Gradle features can be enabled or disabled via <a href="https://docs.gradle.org/current/userguide/build_environment.html">build options</a>,
+ * but in some cases other flags or features can affect that.
+ * For this reason, the {@link #getRequested()} status represents the configuration of a feature in isolation,
+ * and the {@link #getActive()} status represents the final result, after everything has been taken into account.
+ * <p>
+ * Some flags disable features.
+ * For instance, {@code --export-keys} flag always disables {@link BuildFeatures#getConfigurationCache() Configuration Cache}.
+ * If {@code --configuration-cache} is provided at the same time, then {@code requested == true}, but {@code active == false}.
```

Comment:
❓  

Shouldn't be:

`{@code configurationCache.requested == true}` ?

Same for `active`. You did that for the example below, and it was clearer.

Reply:
I thought in this paragraph it can be shorter, because no other features are involved. But I agree that making it more explicit wouldn't hurt